adjust readme for v2 release (#612)
SalesforceCommerceCloud/pwa-kit: packages/template-retail-react-app/README.md

> # The Retail React App 🛒

Reviewer: ```suggestion
# The Retail React App
```

For consistency with other README files in our repo, use emoji only on h2s (and put the emoji before the heading text, not after).

### Dedicated Configuration Files

At project generation a single `default` configuration is created. For most cases a single configuration file will get the job done. But there are times when you want to have a different configuration for a different environment. These environments for example could be, a developers local machine, a remote `production` environment, or a remote `staging` environment. In all these cases you want to use a different configuration for each. 

> When multiple configurations are present, which configuration file to used is resolved in the following way: 

Reviewer: ```suggestion
When multiple configurations are present, here's how the correct configuration file is chosen: 
```

1. Check to see if there is a config file named after your remote Managed Runtime environment and use this if one is present.
2. If we are running on a developer machine, look for a `local` configuration and use this if one is present.
3. Load the `default` config file `config/default.js`.

> This allows you to do these but are not limited to things:

Reviewer: ```suggestion
This configuration system enables the following use cases:
```

> This configuration system allows you to do the following:

Reviewer: ```suggestion
This configuration system allows you to:
```

- Each developer can have their own configuration, connection to their own Commerce API sandboxes.

> - Deploy a single codebase application to multiple environments with their own specific configurations (Multiple B2C Site with Different Domains).

Reviewer: ```suggestion
- Deploy a single PWA Kit codebase to multiple environments, each with their own configuration.
```

Author: Previously we have `This configuration system allows you to:`, it makes sense to start this sentence with `deploy...`

Reviewer: Sure, we can do that but we need to make both list items parallel (consistent sentence structure).

Author: Anw, let's not too caught up with this section here. We can always fix it up later on. I will stop pushing any change and wait for the CI to merge this PR . Thank you for all the feedback @rsexton404!

Reviewer: Scratch that. I'm just going to use the text we already wrote for the docs instead.

    }
}
```

> You can choose how the current locale appears (or does not appear) in the URL by setting `url.locale` to one of the following values:

Reviewer: ```suggestion
You can choose how the current locale appears (or doesn't appear) in the URL by setting `url.locale` to one of the following values:
```
^Salesforce style: use contractions (like "doesn't") whenever possible.
